### ISSUE TITLE
Clicking g-link offline, then clicking it again online should proceed with navigation instead of keeping to fail

### ISSUE THREAD

**Issue by khalyomede** [gridsome/gridsome#1312]
## Summary

When clicking a `<g-link>`, while being offline, a call to graphql data fails. When the network is back, clicking this same link again still does not proceed by navigating to the requested view. In the last section, [Notes](#notes), I show 2 errors that seems to be involved.

### Basic example

1. Grab a fresh new Gridsome project
2. In your developer console, switch the network to "Offline"
3. Click the "About" link
4. It fails to load (with errors in console)
5. In your developer console, switch back the network to "Online"
6. Click again on the "About" link
7. Still no navigation is proceeded

### Motivation

I am creating a web app where this kind of network fluctuations is excpected to happen a lot. Having the ability to proceed to navigation when the network comes back would make the application more reliable no matter the network changes.

### Notes

Here is the errors that are raised when clicking a `<g-link>` while offline:

_fetch.js?c9c9:22_
> POST http://localhost:3000/___graphql net::ERR_INTERNET_DISCONNECTED

_fetch.js?c9c9:33_
> Uncaught (in promise) TypeError: Cannot read property 'errors' of undefined

```javascript
import prefetch from './utils/prefetch'
import { unslashEnd } from './utils/helpers'
import { NOT_FOUND_PATH } from '~/.temp/constants'

const dataUrl = process.env.DATA_URL
const isPrefetched = {}
const isLoaded = {}

export default (route, options = {}) => {
  const { shouldPrefetch = false, force = false } = options

  if (!process.isStatic) {
    const { dynamic = false } = route.meta
    let path = dynamic ? route.matched[0].path : route.path

    if (route.name === '*') {
      path = NOT_FOUND_PATH
    }

    return new Promise((resolve, reject) => {
      if (force || !isLoaded[route.path]) {
        isLoaded[route.path] = fetch(process.env.GRAPHQL_ENDPOINT, { // Failed to load resource: net::ERR_INTERNET_DISCONNECTED
          method: 'POST',
          headers: { 'Content-Type': 'application/json' },
          body: JSON.stringify({ path, dynamic })
        })
          .then(res => res.json())
          .catch(reject)
      }

      isLoaded[route.path]
        .then(res => {
          if (res.errors) reject(res.errors[0]) // Uncaught (in promise) TypeError: Cannot read property 'errors' of undefined
          else if (res.code) resolve({ code: res.code })
          else resolve({
            data: res.data,
            context: res.extensions
              ? res.extensions.context
              : {}
          })

          isLoaded[route.path] = null
        })
    })
  }
```

When clicking the link being online again, only the second error, `Uncaught (in promise) TypeError: Cannot read property 'errors' of undefined` keeps raising, but not the first.

**Comment by hjvedvik** [comment]
I see this also happens in production if the server is shut down while navigating the site. Gridsome doesn't clear the promise in `isLoaded` if the request fails. I'll fix it.

**Comment by hjvedvik** [comment]
@khalyomede I'm closing this since it should be fixed in v0.7.20, but feel free to open it again if you still have issues :)
